startup.sh

```diff
@@ -189,8 +189,51 @@ if [ ! "${ENABLE_DEFAULT_SHUTDOWN}" = "true" ]; then
   REPLACEMENT=
 fi
 
-if [ -n "$GEOSERVER_ADMIN_PASSWORD" ] && [ -n "$GEOSERVER_ADMIN_USER" ]; then
-    /bin/sh /opt/update_credentials.sh
+# Configure GeoServer admin credentials
+# Supports: GEOSERVER_ADMIN_USER/PASSWORD (env vars) or GEOSERVER_ADMIN_USER_FILE/PASSWORD_FILE (file paths)
+# Priority: Direct env vars take precedence over files
+
+ADMIN_USER=""
+ADMIN_PASSWORD=""
+
+# Resolve password from file or env var
+if [ -n "$GEOSERVER_ADMIN_PASSWORD_FILE" ] && [ -f "$GEOSERVER_ADMIN_PASSWORD_FILE" ]; then
+    if [ -n "$GEOSERVER_ADMIN_PASSWORD" ]; then
+        echo "Warning: Both GEOSERVER_ADMIN_PASSWORD and GEOSERVER_ADMIN_PASSWORD_FILE are set. Using GEOSERVER_ADMIN_PASSWORD."
+        ADMIN_PASSWORD="$GEOSERVER_ADMIN_PASSWORD"
+    else
+        ADMIN_PASSWORD=$(cat "$GEOSERVER_ADMIN_PASSWORD_FILE")
+        echo "Loaded GeoServer admin password from file: $GEOSERVER_ADMIN_PASSWORD_FILE"
```

Comment:
From a security perspective it is not recommendable to log file names/paths that contain sensitive information, so I would suggest to do avoid this information here and just log a generic information.

```suggestion
        echo "Loaded GeoServer admin password from file"
```